rs/bitcoin/ckbtc/minter/ckbtc_minter.did

```diff
@@ -193,6 +193,9 @@ type InitArgs = record {
 
     /// The canister id of the KYT canister.
     kyt_principal: opt principal;
+
+    /// The canister id of the new KYT canister.
+    new_kyt_principal: opt principal;
```

Comment:
I'd reuse the same canister, with the same canister ID, rather than introducing a new canister.
What if we want to change the KYT mechanism again in the future? Would that require yet another new canister ID?

Quite generally, I can't see a reason why keeping the KYT canisters separate is useful. If there is a good reason, please let me know!

Reply:
The only reason that I can think of is that things break if the ckBTC minter and the KYT canister are not upgraded together---but I'd still rather vote in both changes at the same time rather than having these extra fields in the code potentially forever.

Reply:
The problem with new/not new is that one doesn't really know what it means and it will get outdated as soon as we use the "new" KYT. It's also also not that easy 
> I can remove the old one

Since we store the upgrade args in the event logs and we have to be able to replay them.

I would be in favour of maybe introducing an enum here, e.g.

```rust
pub enum Kyt {
  Chainanalysis(Principal)
  Ofac(Principal)
}
```
This could be a new field in `lifecycle::UpgradeArgs` and the current field `kyt_principal` should no longer be public (so that one uses the new field). WDYT?

Reply:
That is certainly an option, but it also means that I will need to combine both #2240 and #2304 into a single PR for this to work out. 

Given that existing tests are conditioned on only one kind of KYT canister, it may also mean these tests need to be duplicated to test both, because fee structure, reimbursement etc, will all be different.

I'm not sure if it is worth all the trouble to have the option of switching between KYT canisters with an upgrade flag. What's your opinion on this, @THLO ?

Reply:
Discussed offline: having a transition period during which either the old or the new KYT could be used is not worth the effort. The added field `new_kyt_principal` is temporary and will be removed in a follow-up PR.